pyemma/util/discrete_trajectories.py

```diff
@@ -222,6 +224,163 @@ def number_of_states(dtrajs, only_used = False):
 # indexing
 ################################################################################
 
+def milestone_counting(dtrajs, core_set, in_place=False):
+    r""" Perform milestone counting in given discrete trajectories between core sets.
+
+    The given discrete trajectories are rewritten such, that only transitions between core sets
+    are left in the result.
+
+    Examples
+    --------
+    Lets assume we want to restrict the core sets to 1, 2 and 3:
+
+    >>> import numpy as np
+    >>> dtrajs = [np.array([5, 4, 1, 3, 4, 4, 5, 3, 0, 1]),
+    ...           np.array([4, 4, 4, 5]),
+    ...           np.array([4, 4, 5, 1, 2, 3])]
+    >>> dtraj_core, offsets, n_cores = milestone_counting(dtrajs, core_set=[0, 1, 3])
+    >>> print(dtraj_core)
+    [array([1, 3, 3, 3, 3, 3, 0, 1]), array([1, 1, 3])]
+
+    We reach the first mile stone in the first trajectory after two steps, after four in the second and so on:
+    >>> print(offsets)
+    [2, None, 3]
+
+    Since the second trajectory never visited a core set, it will be removed and marked as such in the offsets
+    lists by a 'None'. Each entry corresponds to one entry in the input list.
+
+    Parameters
+    ----------
+    dtrajs: array_like or list of array_like
+        Discretized trajectory or list of discretized trajectories.
+
+    core_set: None or ndarray(dtype=int) or list of ndarrays(dtype=int)
+        * If None is given, all indices except -1 are considered to be core sets.
+        * Pass an array of micro-states to define the core sets.
+
```

Comment:
From this, I don't understand what happends if "list of ndarrays(dtype=int)" is passed. Also not sure if this case is handled as I would guess...